### PR TITLE
Make 'validate_onnx' function robust against the models loaded without external data

### PR DESCRIPTION
**Pull request by dbogunowicz**
Accounting for the scenario, where the in-memory model passed to the `validate_onnx` is loaded without the external data.
Right now, the validator simply fails. What it should do, it should skip the validation and emit a verbose warning - this is what has been implemented.

Testing:
```python
import onnx
from sparsezoo.utils.onnx import validate_onnx
model = onnx.load("/network/damian/sparsegpt_webinar_fp32/sparsegpt_1.3b/model.onnx", load_external_data=False)
validate_onnx(model)
model_ = onnx.load("/network/damian/sparsegpt_webinar_fp32/sparsegpt_1.3b/model.onnx", load_external_data=True)
validate_onnx(model_)
```

```bash
Attempting to validate an in-memory ONNX model that has been loaded without external data. This is currently not supported by the ONNX checker. The validation will be skipped.
Attempting to validate an in-memory ONNX model with size > 2000000000 bytes.`validate_onnx` skipped, as large ONNX models cannot be validated in-memory. To validate this model, save it to disk and call `validate_onnx` on the file path.
```

### CODE REVIEW

File: src/sparsezoo/utils/onnx.py [hunk 1]
```diff
@@ -50,6 +51,30 @@
...
+    (they are not empty)
+
+    :param model: the ModelProto to check
+    :return True if the model was loaded with external data, False otherwise.
+    """
+
+    initializers = model.graph.initializer
+
+    data_saved_to_disk = any(
+        [initializer.external_data for initializer in initializers]
```
Comment (rahul-tuli) [inline src/sparsezoo/utils/onnx.py:70]:
```suggestion
        initializer.external_data for initializer in initializers
```
nit: let's take advantage of generators here?

File: src/sparsezoo/utils/onnx.py [hunk 2]
```diff
@@ -121,6 +146,14 @@ def validate_onnx(model: Union[str, ModelProto]):
             return
         onnx.checker.check_model(onnx_model)
     except Exception as err:
```
Comment (rahul-tuli) [inline src/sparsezoo/utils/onnx.py:148]: Thought: This seems like a very broad except clause, could we shrink it down to a fewer more specific exceptions maybe?

**Review by rahul-tuli — COMMENTED** [review]
A few suggestions, but the diff looks real nice!

**Review by KSGulin — COMMENTED** [review]
LGTM pending Rahul's comments

File: src/sparsezoo/utils/onnx.py (hunk 2 above)
Comment (dbogunowicz) [reply src/sparsezoo/utils/onnx.py:148]: Maybe, but I do not think we should work on this now, may be an act of overoptimization.

**Review by KSGulin — APPROVED** [review]

File: src/sparsezoo/utils/onnx.py (hunk 2 above)
Comment (rahul-tuli) [reply src/sparsezoo/utils/onnx.py:147]: Still not convinced, but giving a green tick

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**